tests/suites/target_test.function

```diff
@@ -59,10 +59,29 @@ int verify_dependencies( uint8_t count, uint8_t * dep_p )
     return( DEPENDENCY_SUPPORTED );
 }
 
+/**
```

Comment:
Why does the data encoding need to change? Why not fix the Python script to work with binary data correctly?

Reply:
I tried it, but python 2.7 and python 3.7 behave differently with the decoding.
Specifically, when trying to generate the hex string,we get issues, as we need to supply encoding in python3, and since there are bytes that are not part of encoding, the data is wrong. 
See https://github.com/ARMmbed/mbedtls/pull/2660#discussion_r289874079 as example

Reply:
Yes, you need to declare encodings properly. Python 2 is often a pain because a string is just a list of characters and you silently get wrong data if you pass that string to a function that expects a string in a specific encoding. Python 3 is often a pain because there are two types of strings but it isn't always clear what functions expect. But all this means is that the Python script needs to be fixed. Why would you change the communication protocol? Both Python languages are definitely capable of manipulating binary data.

Reply:
yes, but they represent the binary data differently
In python2 it is just bunary data, and in python 3 it is b'....'
If we want same output for both versions, we need to change the communication.

Reply:
They represent the data differently internally, but we don't care. If the script doesn't emit the desired format, it's a bug in the script. There's no need to change the protocol. Just fix the bug.

Reply:
It's not internally. It's the way the bytes are being sent to the target

Reply:
Exactly. If the script sends the wrong bytes to the target, that's a bug in the script.

Reply:
but sending the same binary data on both python3 and python2  u8siong same code can't be done. At least not as far as I seen. This is why I changed the format of the connection

Reply:
It can be done. I can help if you tell me exactly how to run this script so I can test what I'm doing.

Reply:
We chatted about this and concluded that the problem is not the python version, but some evolution in greentea or one of the libraries it uses, which is now explained in the commit message.